src/core/layout/qgslayoutitempicture.cpp

```diff
@@ -355,7 +355,10 @@ void QgsLayoutItemPicture::refreshPicture( const QgsExpressionContext *context )
     {
       mHasExpressionError = true;
       source = QString();
-      QgsMessageLog::logMessage( tr( "Picture expression eval error" ) );
+      if ( scopedContext.feature().isValid() )
```

Comment:
Why this check?

Reply:
@nyalldawson , because otherwise you get a bunch of meaningless errors when loading a project.

What happens there is that during project load, the layouts are loaded, but the atlas feature not set yet. In turn, any image item with a data-defined property that refers to an atlas field (say `'/my/images/' || "PHOTO"`) throws a warning.

Reply:
@nyalldawson , i.e. this:

![image](https://user-images.githubusercontent.com/1728657/222644184-324f97d7-0b7d-4200-a52f-3481616308e7.png)

Reply:
Fair enough -- I just wonder about expressions which don't depend on feature fields though...

Reply:
@nyalldawson , yeah it's not ideal but that message spamming the logs is really not ideal. I was thinking of removing it altogether TBH

Reply:
Ok, agreed